### PR TITLE
Code smell reduction

### PR DESCRIPTION
**Pull request by matteobaccan**
I have corrected some code smell

- Correct use of <> diamond operator
- Correct modifiers order
- Private constructor for some utility class
- Correct use of diamond operator
- Corrected naming convention
- Correct modifier order
- Use not synchronized class
- Introduced try/resource in stream copy
- Removed unused private field 

I hope this may be usefull for Apktool

### CODE REVIEW

File: brut.j.dir/src/main/java/brut/directory/DirUtil.java [hunk 1]
```diff
@@ -28,6 +28,10 @@
 public class DirUtil {
     private static final Logger LOGGER = Logger.getLogger("");
 
+    private DirUtil(){
```
Comment (iBotPeaches) [inline brut.j.dir/src/main/java/brut/directory/DirUtil.java:31]: Do you mind explaining what benefit the no-op constructors have? As well as little space here `private DirUtil() {`

File: brut.j.dir/src/main/java/brut/directory/ZipUtils.java [hunk 2]
```diff
@@ -30,6 +30,10 @@
 public class ZipUtils {
 
     private static Collection<String> mDoNotCompress;
+    
+    private ZipUtils(){
```
Comment (iBotPeaches) [inline brut.j.dir/src/main/java/brut/directory/ZipUtils.java:34]: `private ZipUtils() {`

File: brut.j.util/src/main/java/brut/util/OS.java [hunk 3]
```diff
@@ -72,11 +72,11 @@ public static void cpdir(File src, File dest) throws BrutException {
...
-                InputStream in = new FileInputStream(file);
-                OutputStream out = new FileOutputStream(destFile);
-                IOUtils.copy(in, out);
-                in.close();
-                out.close();
+                try(InputStream in = new FileInputStream(file)){
+                    try(OutputStream out = new FileOutputStream(destFile)){
+                        IOUtils.copy(in, out);               
+                    }
+                }
```
Comment (iBotPeaches) [inline brut.j.util/src/main/java/brut/util/OS.java:79]:
```
try (InputStream in = new FileInputStream(file)) {
    try (OutputStream out = new FileOutputStream(destFile)) {
        IOUtils.copy(in, out);               
    }
}
```

Suggested spacing fixes

File: brut.j.util/src/main/java/brut/util/OS.java [hunk 4]
```diff
@@ -173,7 +173,7 @@ public void run() {
     }
 
     static class StreamCollector implements Runnable {
-        private final StringBuffer buffer = new StringBuffer();
+        private final StringBuilder buffer = new StringBuilder();
```
Comment (iBotPeaches) [inline brut.j.util/src/main/java/brut/util/OS.java:176]: I though I wanted a StringBuffer to properly block/handle multiple threads capturing output.

**Review by iBotPeaches — COMMENTED** [review]
Thanks! Some style fixes and comments about some of the fixes.

File: brut.j.dir/src/main/java/brut/directory/DirUtil.java (hunk 1 above)
Comment (matteobaccan) [reply brut.j.dir/src/main/java/brut/directory/DirUtil.java:31]: This is a Java best practice raccomendation
http://www.javapractices.com/topic/TopicAction.do?Id=40

File: brut.j.dir/src/main/java/brut/directory/ZipUtils.java (hunk 2 above)
Comment (matteobaccan) [reply brut.j.dir/src/main/java/brut/directory/ZipUtils.java:34]: This is a Java best practice recommendation
http://www.javapractices.com/topic/TopicAction.do?Id=40

File: brut.j.util/src/main/java/brut/util/OS.java (hunk 3 above)
Comment (matteobaccan) [reply brut.j.util/src/main/java/brut/util/OS.java:79]: Sorry, I don't have reformat the code.
I must reformat and resubmit a pull request?

Comment (iBotPeaches) [reply brut.j.util/src/main/java/brut/util/OS.java:79]: Just another commit is fine, no need for another PR

Comment (matteobaccan) [reply brut.j.util/src/main/java/brut/util/OS.java]: I have update the PR with a new commit

File: brut.j.dir/src/main/java/brut/directory/ZipUtils.java (hunk 2 above)
Comment (ThexXTURBOXx) [reply brut.j.dir/src/main/java/brut/directory/ZipUtils.java:34]: Yes, but there is a space missing ;)

Comment (matteobaccan) [reply brut.j.dir/src/main/java/brut/directory/ZipUtils.java]: I have add a commit with the space.

File: brut.j.dir/src/main/java/brut/directory/DirUtil.java (hunk 1 above)
Comment (matteobaccan) [reply brut.j.dir/src/main/java/brut/directory/DirUtil.java]: I have add a commit with the space.

File: brut.j.util/src/main/java/brut/util/OS.java [hunk 5]
```diff
@@ -188,7 +188,7 @@ public void run() {
                 while ((line = reader.readLine()) != null) {
                     buffer.append(line).append('\n');
                 }
-            } catch (IOException ignored) {}
+            } catch (IOException ignored) {}
```
Comment (iBotPeaches) [inline brut.j.util/src/main/java/brut/util/OS.java:191]: final comment - there is some extra spacing added to this line. Can it be removed?

Comment (matteobaccan) [reply brut.j.util/src/main/java/brut/util/OS.java]: Removed ;)

**Review by iBotPeaches — APPROVED** [review]

**Comment by iBotPeaches** [comment]
thanks!

**Pull request merged**